modules/javafx.graphics/src/main/java/javafx/stage/PopupWindow.java

```diff
@@ -479,6 +477,23 @@ private void showImpl(final Window owner) {
         }
     }
 
+    /**
+     * Applies the stylesheet from the scene of the root owner {@link Window} to the {@link Scene}
+     * associated with that window.
+     *
+     * @param owner the owner {@link Window}
+     */
+    protected void applyStylesheetFromOwner(Window owner) {
+        Scene scene = getScene();
+        final Scene ownerScene = getRootWindow(owner).getScene();
+        if (ownerScene != null) {
+            if (ownerScene.getUserAgentStylesheet() != null) {
+                scene.setUserAgentStylesheet(ownerScene.getUserAgentStylesheet());
+            }
+            scene.getStylesheets().setAll(ownerScene.getStylesheets());
```

Comment:
so if the application decides to set a particular style sheet on the tooltip scene, it will be ignored?  Should it be addAll()?  And if yes, should it check against duplicate entries?

Reply:
Yes, but I did not change this behaviour, as this method is always called right before showing.
While this a bit weird, this is out of scope for at least this PR.